deploy/utils/validate_deployments.py

```
@@ -0,0 +1,422 @@
+
+        return "".join(parts)
+
+
+def has_dgd_resource(yaml_file: Path) -> bool:
+    """Check if a YAML file contains DynamoGraphDeployment resources."""
+    try:
+        with open(yaml_file) as f:
+            for doc in yaml.safe_load_all(f):
+                if doc and doc.get("kind") == "DynamoGraphDeployment":
+                    return True
+        return False
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)